saleor/webhook/transport/synchronous/circuit_breaker/breaker_board.py

```diff
@@ -20,25 +20,37 @@ class BreakerBoard:
     webhook execution.
     """
 
-    def __init__(self, storage: Storage, failure_threshold: int):
+    def __init__(
+        self,
+        storage: Storage,
+        failure_threshold: int,
+        failure_min_count: int,
+        cooldown_seconds: int,
+        ttl: int,
+    ):
         self.storage = storage
 
-        # TODO - make everything below configurable
-        self.failure_threshold = failure_threshold
+        # TODO - make event types configurable
         self.webhook_event_types = [
             WebhookEventSyncType.SHIPPING_LIST_METHODS_FOR_CHECKOUT
         ]
-        self.cooldown_seconds = 5 * 60  # 5 minutes
-        self.ttl = 10 * 60  # 10 minutes
+        self.failure_threshold = failure_threshold
+        self.failure_min_count = failure_min_count
+        self.cooldown_seconds = cooldown_seconds
+        self.ttl = ttl
 
     def is_closed(self, app_id: int):
         # TODO - cache last open to optimize?
         return self.storage.last_open(app_id) < (time.time() - self.cooldown_seconds)
 
     def register_error(self, app_id: int):
         errors = self.storage.register_event_returning_count(app_id, "error", self.ttl)
+        total = self.storage.register_event_returning_count(app_id, "total", self.ttl)
+
+        if total < self.failure_min_count:
+            return
 
-        if errors >= self.failure_threshold:
+        if (errors / total) * 100 >= self.failure_threshold:
```

Comment:
Shouldn't we also confirm that `total != 0`?

Reply:
Not possible as the state is already incremented in the storage.